### PR TITLE
Add destination query parameter to job full logs

### PR DESCRIPTION
**Pull request by fviale**

### CODE REVIEW

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java [hunk 1]
```diff
@@ -885,8 +892,17 @@ public InputStream jobFullLogs(String sessionId, String jobId, String session) t
...
-            // will produce HTTP 204 code if null
-            return streams.isEmpty() ? null : new SequenceInputStream(Collections.enumeration(streams));
+            builder = Response.ok()
+                              .entity(streams.isEmpty() ? IOUtils.toInputStream("",
+                                                                                Charset.forName(PASchedulerProperties.FILE_ENCODING.getValueAsString()))
+                                                        : new SequenceInputStream(Collections.enumeration(streams)));
+
+            if (DESTINATION_FILE.equals(destination)) {
+                builder = builder.header(HttpHeaders.CONTENT_DISPOSITION,
+                                         "attachment; filename=job_" + jobId + "_logs.txt");
```
Comment (amouhoub) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:902]: shouldn't the extension be .log for more convenience and easy opening with log monitoring software ?

File: rest/rest-server/src/test/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRestJobLogsTest.java [hunk 2]
```diff
@@ -105,9 +107,11 @@ public void job_full_logs_not_finished() throws Exception {
         jobState.addTask(new InternalScriptTask(jobState));
         when(mockScheduler.getJobState("123")).thenReturn(jobState);
 
-        InputStream fullLogs = restScheduler.jobFullLogs(validSessionId, "123", validSessionId);
+        InputStream fullLogs = (InputStream) restScheduler.jobFullLogs(validSessionId, "123", validSessionId, null)
```
Comment (amouhoub) [inline rest/rest-server/src/test/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRestJobLogsTest.java:110]: @fviale  Wouldn't it be useful to add a test alternating the sessionId parameter (one in the header & one in the query params) ?

Comment (fviale) [reply rest/rest-server/src/test/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRestJobLogsTest.java:110]: ok

Comment (fviale) [reply rest/rest-server/src/test/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRestJobLogsTest.java:110]: I don't think this test would be useful

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java (hunk 1 above)
Comment (fviale) [reply rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:902]: ok

**Review by AlexFalessi — APPROVED** [review]

**Review by amouhoub — APPROVED** [review]

**Pull request merged**